### PR TITLE
Snap Docker image with preloaded plugins

### PR DESCRIPTION
**Pull request by mkuculyma**
Docker image with plugins and tasks to monitor Kubernetes. No tribe used.
Plugins used in this image:
- InfluxDB publisher,
- CPU collector,
- Docker collector,
- Kubestate collector,
- Snapstats collector,
- Meminfo collector

Tested manually.

### CODE REVIEW

File: src/helm/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+FROM ubuntu:16.04
+MAINTAINER Maria Rolbiecka maria.a.rolbiecka@intel.com
+
+RUN apt-get update && apt-get install -y curl stress
```
Comment (andrzej-k) [inline src/helm/Dockerfile]: Do we really need 'stress'?

File: src/helm/Dockerfile [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+COPY start.sh /
+ENTRYPOINT ["/start.sh"]    
+    
+    
+    
+    
+    
+    
+    
+
```
Comment (andrzej-k) [inline src/helm/Dockerfile]: Those empty lines can be probably removed

File: src/helm/Dockerfile [hunk 3]
```diff
@@ -0,0 +1,67 @@
+#
+# http://www.apache.org/licenses/LICENSE-2.0.txt
+# Copyright 2016 Intel Corporation
```
Comment (andrzej-k) [inline src/helm/Dockerfile]: 2016 -> 2017

File: src/helm/snapteld.conf [hunk 4]
```diff
@@ -0,0 +1,118 @@
+# Copyright 2016 Intel Corporation
```
Comment (andrzej-k) [inline src/helm/snapteld.conf]: 2016 -> 2017

File: src/helm/snapteld.conf [hunk 5]
```diff
@@ -0,0 +1,118 @@
...
+
+# log_path sets the path for logs for the snap daemon. By
+# default snapteld prints all logs to stdout. Any provided
+# path will send snapteld logs to a file called snapteld.log in
+# the provided directory.
+# log_path: /snap
+
+# Gomaxprocs sets the number of cores to use on the system
+# for snapteld to use. Default for gomaxprocs is 1
+# gomaxprocs: 1
```
Comment (andrzej-k) [inline src/helm/snapteld.conf]: Is single core enough to collect that much metrics? We may need to allow users to assign more cores, for example via chart config.

File: src/helm/start.sh [hunk 6]
```diff
@@ -0,0 +1,17 @@
+#!/bin/bash
```
Comment (andrzej-k) [inline src/helm/start.sh]: license header is missing

File: src/helm/tasks/docker.yml [hunk 7]
```diff
@@ -0,0 +1,27 @@
...
+  name: docker
+  schedule:
+    type: "simple"
+    interval: "10s"
+    deadline: "10s"
+  max-failures: -1
+  workflow:
+    collect:
+      metrics:
+        /intel/procfs/*: {}
```
Comment (andrzej-k) [inline src/helm/tasks/docker.yml]: Can we constrain list of collected metrics to only those needed to be reported:

- file/cpu/memory utilization for hosts
- File/cpu/memory utilization for containers
- Disk utilization for for persistent disks

File: src/helm/tasks/kubestate.yml [hunk 8]
```diff
@@ -0,0 +1,26 @@
...
+  name: kubestate
+  schedule:
+    type: "simple"
+    interval: "10s"
+    deadline: "10s"
+  max-failures: -1
+  workflow:
+    collect:
+      metrics:
+        /grafanalabs/kubestate/*: {}
```
Comment (andrzej-k) [inline src/helm/tasks/kubestate.yml]: Can we only collect metrics that are not already collected by docker plugin?

File: src/helm/Dockerfile [hunk 9]
```diff
@@ -0,0 +1,67 @@
...
+	&& echo "${DOCKER_SHA256}  /usr/local/bin/docker" | sha256sum -c - \
+	&& chmod +x /usr/local/bin/docker
+
+RUN mkdir -p /var/log/snap /opt/snap/autoload /proc_host
+
+RUN cd /opt/snap/autoload \
+    && curl -fSL "https://github.com/raintank/snap-plugin-collector-snapstats/releases/download/v0.0.1/snap-plugin-collector-snapstats" -o snap-plugin-collector-snapstats \
+    && curl -fSL "https://github.com/intelsdi-x/snap-plugin-publisher-influxdb/releases/download/21/snap-plugin-publisher-influxdb_linux_x86_64" -o snap-plugin-publisher-influxdb \
+    && curl -fSL "https://github.com/intelsdi-x/snap-plugin-collector-cpu/releases/download/6/snap-plugin-collector-cpu_linux_x86_64" -o snap-plugin-collector-cpu \
+    && curl -fSL "https://github.com/intelsdi-x/snap-plugin-collector-meminfo/releases/download/3/snap-plugin-collector-meminfo_linux_x86_64" -o snap-plugin-collector-meminfo \
```
Comment (andrzej-k) [inline src/helm/Dockerfile]: Why meminfo and cpu are needed?

File: src/helm/tasks/procfs.yml [hunk 10]
```diff
@@ -0,0 +1,26 @@
...
+      - plugin_name: "influxdb"
+        config:
+          host: "monitoring-influxdb"
+          port: 8086
+          database: "snap"
+          user: admin 
+          password: admin
+          scheme: http
+          skip-verify: true 
+          isMultiFields: false
```
Comment (andrzej-k) [inline src/helm/tasks/procfs.yml]: Do we need this task?

File: src/helm/snapteld.conf (hunk 5 above)
Comment (mkuculyma) [inline src/helm/snapteld.conf]: ok

File: src/helm/start.sh (hunk 6 above)
Comment (mkuculyma) [inline src/helm/start.sh]: added

**Pull request merged**